src/content/subgames.cpp

```diff
@@ -210,40 +222,46 @@ SubgameSpec findWorldSubgame(const std::string &world_path)
 	return findSubgame(world_gameid);
 }
 
-std::set<std::string> getAvailableGameIds()
+GamePathMap getAvailableGamePaths()
 {
-	std::set<std::string> gameids;
-	std::set<std::string> gamespaths;
-	gamespaths.insert(porting::path_share + DIR_DELIM + "games");
-	gamespaths.insert(porting::path_user + DIR_DELIM + "games");
+	GamePathMap gamepaths;
+	std::vector<GameFindPath> gamespaths{
+		{porting::path_share + DIR_DELIM + "games", false},
+		{porting::path_user + DIR_DELIM + "games", true}
+	};
 
 	Strfnd search_paths(getSubgamePathEnv());
 
 	while (!search_paths.at_end())
-		gamespaths.insert(search_paths.next(PATH_DELIM));
+		gamespaths.emplace_back(search_paths.next(PATH_DELIM), false);
 
-	for (const std::string &gamespath : gamespaths) {
-		std::vector<fs::DirListNode> dirlist = fs::GetDirListing(gamespath);
+	for (const GameFindPath &gamespath : gamespaths) {
+		std::vector<fs::DirListNode> dirlist = fs::GetDirListing(gamespath.path);
 		for (const fs::DirListNode &dln : dirlist) {
 			if (!dln.dir)
 				continue;
 
 			// If configuration file is not found or broken, ignore game
 			Settings conf;
-			std::string conf_path = gamespath + DIR_DELIM + dln.name +
+			std::string conf_path = gamespath.path + DIR_DELIM + dln.name +
 						DIR_DELIM + "game.conf";
 			if (!conf.readConfigFile(conf_path.c_str()))
 				continue;
 
 			// Add it to result
-			const char *ends[] = {"_game", NULL};
-			auto shorter = removeStringEnd(dln.name, ends);
-			if (!shorter.empty())
-				gameids.emplace(shorter);
-			else
-				gameids.insert(dln.name);
+			gamepaths[normalizeGameId(dln.name)]
+				= {gamespath.path + DIR_DELIM + dln.name, gamespath.user_specific};
 		}
 	}
+	return gamepaths;
+}
+
+std::set<std::string> getAvailableGameIds()
+{
+	GamePathMap gamepaths = getAvailableGamePaths();
+	std::set<std::string> gameids;
+	for (auto &&p : gamepaths)
```

Comment:
```suggestion
	for (const auto &p : gamepaths)
```

Reply:
Doesn't an r-value reference make more sense here?

Reply:
if you want to avoid copying the string you will have to use std::move